app/controllers/admin/changeVolunteer.py

```diff
@@ -0,0 +1,51 @@
+from flask import request, render_template, redirect, url_for, request, flash
+from app.controllers.admin import admin_bp
+from app.models.event import Event
+from app.models.user import User
+from app.models.eventParticipant import EventParticipant
+from app.logic.searchVolunteers import searchVolunteers
+from app.logic.updateVolunteers import updateVolunteers, addVolunteerToEvent
+from app.models.user import User
+from peewee import *
+
+@admin_bp.route('/searchVolunteers/<query>', methods = ['GET'])
+def getVolunteers(query):
+    '''Accepts user input and queries the database returning results that matches user search'''
+    query = query.strip()
+    search = query.upper()
+    splitSearch = search.split()
+
+    return searchVolunteers(query)
+
+
+
+@admin_bp.route('/<programID>/<eventID>/track_volunteers', methods=['POST'])
+def updateVolunteerTable(programID, eventID):
+
+    volunteerUpdated = updateVolunteers(request.form)
+    if volunteerUpdated:
+        flash("Volunteer table succesfully updated")
+    else:
+        flash("Error adding volunteer")
+
+    return redirect(url_for("admin.trackVolunteersPage", programID=programID, eventID=eventID))
+
+
+@admin_bp.route('/addVolunteerToEvent/<user>/<volunteerEventID>/<eventLengthInHours>', methods = ['POST'])
+def addVolunteer(user, volunteerEventID, eventLengthInHours):
+    user = user.strip("()")
+    userName=user.split('(')[-1]
+    succesfullyAddedVolunteer = addVolunteerToEvent(userName, volunteerEventID, eventLengthInHours)

```

Comment:
I think you should get a user object either here or at the beginning of addVolunteerToEvent so you can validate that the user is an existing user before you try to do anything with the username.